### ISSUE TITLE
sort broken for finder with no critera

### ISSUE THREAD

**Issue by chrisbitmead** [grails/grails-core#11657]
### Steps to Reproduce

Using a finder with no criteria, the sort criteria doesn't work. e.g.

Booking.findAll([sort: ['email': 'desc', 'lastUpdated': 'desc']])

the sort criteria here has no effect. By contrast, if I do:

Booking.findAllByVersion(0, [sort: ['email': 'desc', 'lastUpdated': 'desc']])

the sort criteria works as expected

### Expected Behaviour

Even with no criteria, findAll should sort properly.

### Actual Behaviour

No sorting occurs.

### Environment Information

- **Operating System**: Windows
- **Grails Version:** 4.1.0.M2
- **JDK Version:** 1.8.0_271 Oracle

**Comment by xpusostomos** [comment]
Well... if I'm passing a Map as the first parameter, as far this documentation is concerned, surely I'd be calling this one:
```
Book.findAll(Map queryParams, Closure whereCriteria)
```
And since queryParams is defined as 

And does not the documentation have a findAll with sort as the first parameter: (albeit with a closure)
```
def results = Person.findAll(sort:"firstName") {
     lastName == "Simpson"
}
```
My question is, if ```Booking.findAll([sort: ['email': 'desc', 'lastUpdated': 'desc']])``` doesn't mean what it looks like it means, what does it mean? I mean, there's no runtime error, no compile error, so what does findAll with first argument map mean?

**Comment by jeffbrown** [comment]
> if I'm passing a Map as the first parameter, as far this documentation is concerned, surely I'd be calling this one: `Book.findAll(Map queryParams, Closure whereCriteria)`

I don't think that is the method you are invoking when you run `Booking.findAll([sort: ['email': 'desc', 'lastUpdated': 'desc']])`.

`Book.findAll(Map queryParams, Closure whereCriteria)` would invoke https://github.com/grails/grails-data-mapping/blob/1d5f9fbe7605cb0956276257c5893f4c846cfc32/grails-datastore-gorm/src/main/groovy/org/grails/datastore/gorm/GormEntity.groovy#L476-L478.

Since `Booking.findAll([sort: ['email': 'desc', 'lastUpdated': 'desc']])` is not providing a `Closure`, that should invoke https://github.com/grails/grails-data-mapping/blob/1d5f9fbe7605cb0956276257c5893f4c846cfc32/grails-datastore-gorm/src/main/groovy/org/grails/datastore/gorm/GormEntity.groovy#L690-L692.
